glean/metrics.yaml

```diff
@@ -1467,6 +1467,23 @@ sample:
           The id of the step that was viewed
         type: string
 
+  journal_settings_failure:
+    type: event
+    lifetime: ping
+    send_in_pings:
+      - main
+    description: |
+      We are unable to remove the journal settings file from a rollback.
+    bugs:
+      - https://mozilla-hub.atlassian.net/browse/VPN-3184
+    data_reviews:
+      - TODO
+    data_sensitivity:
+      - technical
+    notification_emails:
+      - amarchesini@mozilla.com
+    expires: never
+
```

Comment:
I think it's good to add telemetry to monitor this functionality. However I wonder if this metric by itself is enough. What questions would we answer with it? We would simply have an absolute count of how many times the journal file fails to be deleted, but nothing else to analyse it against. 

If we are going to add telemetry here I think we would need also events that check:

1. A transaction has started
2. A trasaction has completed succesfully
3. A transaction has failed, with a `reason` extra key on why it failed (which would include this case)

This would allow us to create analyses that check the percentage of failures on this feature and what are the most prominent failures. 

What do you think? Do you think we already have metrics that would help us answer that without adding new ones?

Also, it's fine to do this as a follow up... So I'll r+ this PR for now :)